datarepo-clienttests/src/main/java/common/commands/LockAndRunTest.java

```diff
@@ -0,0 +1,52 @@
+package common.commands;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.TestRunner;
+import runner.config.ServerSpecification;
+import runner.config.TestConfiguration;
+
+public class LockAndRunTest {
+  private static final Logger logger = LoggerFactory.getLogger(LockAndRunTest.class);
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 2) { // if no args specified or invalid number of args specified, print help
+      PrintHelp.printHelp();
+      return;
+    }
+
+    LockNamespace.lockNamespace();
+    unlockShutdownHook();
+
+    // execute a test configuration or suite
+    boolean isFailure;
+    isFailure = TestRunner.runTest(args[0], args[1]);
+    if (isFailure) {
+      System.exit(1);
+    }
+  }
+
+  public static ServerSpecification getServer() throws Exception {
+    // read in the server file
+    String serverEnvVar = TestConfiguration.readServerEnvironmentVariable();
+    if (serverEnvVar == null) {
+      throw new Exception(
+          TestConfiguration.serverFileEnvironmentVarName + " env variable must be defined");
+    }
+    return ServerSpecification.fromJSONFile(serverEnvVar);
+  }
+
+  public static void unlockShutdownHook() {
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
+                () -> {
+                  try {
+                    UnlockNamespace.unlockNamespace();
+                  } catch (Exception e) {
+                    logger.error("Unable to unlock.");
```

Comment:
Should this throw a RunetimeException?  Not sure...I've never worked with shutdown hooks

Reply:
Are you referring to the fact that I'm not actually throwing the exception here?

I'm going back and forth on this.
I was originally thinking it's okay if the unlock fails, as long as it's noted in the log. I don't think we want an entire test run to fail b/c an unlock failed (for example, if it doesn't find the secret b/c it's somehow already been deleted).  But, in theory, it shouldn't fail.

Reply:
Yeah, I think you're probably right.  The process is shutting down anyways